src/makeCalculator.js

```diff
@@ -4,7 +4,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    operate(func, num) {
+      this.result = func(this.result, num);
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    add(res, num) {
+      return res + num;
+    },
+
+    subtract(res, num) {
+      return res - num;
+    },
+
+    divide(res, num) {
+      if (num !== 0) {
+        return res / num;
+      }
+
+      return this.result;
```

Comment:
Returning `this.result` when division by zero occurs might not be the best approach. Consider throwing an error or returning a special value (like `NaN` or `Infinity`) to indicate that the operation is invalid.